tests/test_dependency_cache.py

```diff
@@ -1,9 +1,31 @@
 from fastapi import Depends, FastAPI, Security
+from fastapi._compat import PYDANTIC_V2
 from fastapi.testclient import TestClient
+from pydantic import BaseModel
 
 app = FastAPI()
 
-counter_holder = {"counter": 0}
+counter_holder = {"counter": 0, "parsing_counter": 0}
```

Comment:
IMHO this should be set in pytest session-level context or similar, it shouldn't just be a global variable, I feel that there's too high a risk that the test could end up running with the counter initialised to a non-zero value (e.g. if for some reason the same test runs twice in one pytest invocation, if we end up adding another test in the future that depends on the same counter, etc).